linkerd/app/outbound/src/discover.rs

```diff
@@ -159,9 +159,18 @@ fn spawn_synthesized_profile_policy(
     tokio::spawn(
         async move {
             loop {
-                if profile.changed().await.is_err() {
-                    tracing::debug!("Profile watch closed, terminating");
-                    return;
+                tokio::select! {
+                    biased;
```

Comment:
does this actually need to be biased? AFAICT, it doesn't actually matter whether we poll the closed future or the profile rx first, as either way, we will end up handling the condition --- the important part is just that we are notified by either.

Reply:
yeah that's right. we'll get it in post